src/jit/emit.h

```diff
@@ -1209,6 +1212,99 @@ class emitter
         }
     }; // End of  struct instrDesc
 
+#if defined(_TARGET_XARCH_)
+    insFormat emitter::getMemoryOperation(instrDesc* id);
+#elif defined(_TARGET_ARM64_)
+    void emitter::getMemoryOperation(instrDesc* id, unsigned* pMemAccessKind, bool* pIsLocalAccess);
+#endif
+
+#if defined(DEBUG) || defined(LATE_DISASM)
+
+#define PERFSCORE_THROUGHPUT_ILLEGAL -1024.0f
+
+#define PERFSCORE_THROUGHPUT_4X 0.25f         // Fastest - Quad issue
+#define PERFSCORE_THROUGHPUT_3X (1.0f / 3.0f) // Faster - Three issue
+#define PERFSCORE_THROUGHPUT_2X 0.5f          // Faster - Dual issue
+
+#define PERFSCORE_THROUGHPUT_1C 1.0f // Single Issue
+
+#define PERFSCORE_THROUGHPUT_2C 2.0f   // slower - 2 cycles
+#define PERFSCORE_THROUGHPUT_3C 3.0f   // slower - 3 cycles
+#define PERFSCORE_THROUGHPUT_4C 4.0f   // slower - 4 cycles
+#define PERFSCORE_THROUGHPUT_5C 5.0f   // slower - 5 cycles
+#define PERFSCORE_THROUGHPUT_6C 6.0f   // slower - 6 cycles
+#define PERFSCORE_THROUGHPUT_10C 10.0f // slower - 10 cycles
+#define PERFSCORE_THROUGHPUT_13C 13.0f // slower - 13 cycles
+#define PERFSCORE_THROUGHPUT_25C 25.0f // slower - 26 cycles
```

Comment:
`25.0f` but `26` cycles?

Reply:
Would it be better to use `uint32_t` where `0.25` is `1` and `1.0` is `4`?

`float` is prone to rounding error, especially as the value increases in size; and while I don't think any function will be large enough that it will be a real concern, another model might be better.

Reply:
Yes, the total method perfscore is computed using a double:

` double      compPerfScore;`

These float constants are only used to return the instruction perf characteristics.
THey are widened to double when they contribute to the `compPerfScore`.
I thought about using two enums for `PERFSCORE_THROUGHPUT` and `PERFSCORE_LATENCY`
so that we could pack the `insExecutionCharacteristics` struct to be a 32 or 64 bits.
But this is only used for checked/debug builds and is not a performance impacting area.

Reply:
I don't understand why these defines are needed. They would be useful if they were some broad classification like low/medium/high - you could then easily tune the actual numeric values. But something like "#define CONSTANT_1 1" seems more like unnecessary code bloat.

Reply:
The idea here was that  that in the future these could become  enum values. 
And that we could then pack the insExecutionCharacteristics struct to be a 32 or 64 bits.